plugin/tasks.py

```diff
@@ -13,70 +13,98 @@
-    ctx.logger.info('Create instance')
     gcp = GoogleCloudPlatform(config['auth'],
                               config['project'],
-                              config['scope'])
+                              config['scope'],
+                              ctx.logger)
+
     response = gcp.create_instance(ctx.node.name,
                                    config['agent_image'])
     gcp.wait_for_operation(response['name'])
-    gcp.set_ip()
+    set_ip(gcp)
 
 
 @operation
+@throw_cloudify_exceptions
 def delete_instance(config, **kwargs):
-    ctx.logger.info('Delete instance')
     gcp = GoogleCloudPlatform(config['auth'],
                               config['project'],
-                              config['scope'])
+                              config['scope'],
+                              ctx.logger)
     response = gcp.delete_instance(ctx.node.name)
     gcp.wait_for_operation(response['name'])
 
 
 @operation
+@throw_cloudify_exceptions
 def create_network(config, **kwargs):
-    ctx.logger.info('Create network')
     gcp = GoogleCloudPlatform(config['auth'],
                               config['project'],
-                              config['scope'])
+                              config['scope'],
+                              ctx.logger)
     response = gcp.create_network(config['network'])
     gcp.wait_for_operation(response['name'], True)
 
 
 @operation
+@throw_cloudify_exceptions
 def delete_network(config, **kwargs):
-    ctx.logger.info('Delete network')
     gcp = GoogleCloudPlatform(config['auth'],
                               config['project'],
-                              config['scope'])
+                              config['scope'],
+                              ctx.logger)
     response = gcp.delete_network(config['network'])
     gcp.wait_for_operation(response['name'], True)
 
 
 @operation
+@throw_cloudify_exceptions
 def create_firewall_rule(config, **kwargs):
-    ctx.logger.info('Create instance')
     gcp = GoogleCloudPlatform(config['auth'],
                               config['project'],
-                              config['scope'])
+                              config['scope'],
+                              ctx.logger)
     response = gcp.create_firewall_rule(config['network'], config['firewall'])
     gcp.wait_for_operation(response['name'], True)
 
 
 @operation
+@throw_cloudify_exceptions
 def delete_firewall_rule(config, **kwargs):
-    ctx.logger.info('Create instance')
     gcp = GoogleCloudPlatform(config['auth'],
                               config['project'],
-                              config['scope'])
+                              config['scope'],
+                              ctx.logger)
     response = gcp.delete_firewall_rule(config['network'], config['firewall'])
     gcp.wait_for_operation(response['name'], True)
+
+
+def set_ip(gcp):
+    instances = gcp.list_instances()
+    item = utils.get_item_from_gcp_response('name', ctx.node.name, instances)
```

Comment:
I might be wrong but I think this search will have to change after you merge PR #2 . instead of ctx.node.name you'll be passing the result of utils.get_gcp_resource_name, so it's probably required that before passing it to gcp.create_instance you store it in a runtime property, like" ctx.instance.runtime_properties['hostname'] = your_unique_name. In set_ip, you can read that runtime property to search for the correct instance name.